test/sql/function/timestamp/test_icu_datepart.test

```diff
@@ -284,13 +325,61 @@ SELECT weekday(ts), weekday(ts::TIMESTAMP) FROM timestamps;
 query II
 SELECT weekofyear(ts), weekofyear(ts::TIMESTAMP) FROM timestamps;
 ----
+11	11
+31	31
 53	53
 5	5
 47	47
 46	45
 46	45
 51	51
 
+#
+# Struct parts
+#
+
+# Date parts
+query II
+SELECT DATE_PART(['era', 'year', 'month', 'day'], ts), ts
+FROM timestamps
+ORDER BY 2
+----
+{'era': 0, 'year': 44, 'month': 3, 'day': 15}	0044-03-13 (BC) 09:33:41.987654+00
+{'era': 1, 'year': 1962, 'month': 7, 'day': 31}	1962-07-31 12:20:48.123456+00
+{'era': 1, 'year': 2020, 'month': 12, 'day': 31}	2021-01-01 00:00:00+00
+{'era': 1, 'year': 2021, 'month': 2, 'day': 1}	2021-02-02 00:00:00+00
+{'era': 1, 'year': 2021, 'month': 11, 'day': 15}	2021-11-15 09:30:00+00
+{'era': 1, 'year': 2021, 'month': 11, 'day': 15}	2021-11-15 10:30:00+00
+{'era': 1, 'year': 2021, 'month': 11, 'day': 26}	2021-11-26 10:15:13.123456+00
+{'era': 1, 'year': 2021, 'month': 12, 'day': 24}	2021-12-24 22:00:00+00
+
+# Time parts
+query II
+SELECT DATE_PART(['hour', 'minute', 'microsecond', 'offset'], ts), ts
```

Comment:
Could you add tests for the following scenarios:

* What if the list of parts is empty?
* What about nulls in the list of parts?
* What about non-constants int he list of parts?